### PR TITLE
Generate protobuf classes at compile time.

### PR DESCRIPTION
**Pull request by vitillo**
The externally generated protobuf class wasn't compatible with Flink (it was
throwing some serialization errors). This patch fixes that and also
ensures that a change in the protobuf file doesn't require an external
recompiliation step.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-batch-view/167)

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/heka/package.scala [hunk 1]
```diff
@@ -2,27 +2,27 @@ package com.mozilla.telemetry
...
+      case Field.ValueType.BOOL => f.valueBool(0)
+      case Field.ValueType.DOUBLE => f.valueDouble(0)
+      case Field.ValueType.INTEGER => f.valueInteger(0)
       case _ => ""
     }
   }
 
   implicit def messageToRichMessage(m: Message): RichMessage = new RichMessage(m)
-}
+}
```
Comment (maurodoglio) [inline src/main/scala/com/mozilla/telemetry/heka/package.scala]: missing blank line

File: src/main/scala/com/mozilla/telemetry/heka/HekaFrame.scala [hunk 2]
```diff
@@ -50,7 +50,7 @@ object HekaFrame{
         val uncompressedLength = Snappy.uncompressedLength(messageBuffer)
         val uncompressedMessage = new Array[Byte](uncompressedLength)
         Snappy.uncompress(messageBuffer, 0, header.messageLength, uncompressedMessage, 0)
-        Message.parseFrom(uncompressedMessage, 0, uncompressedLength)
+        Message.parseFrom(uncompressedMessage)
```
Comment (maurodoglio) [inline src/main/scala/com/mozilla/telemetry/heka/HekaFrame.scala]: Is this the difference between the old protobuff library and the new one?

Comment (vitillo) [inline src/main/scala/com/mozilla/telemetry/heka/HekaFrame.scala]: The should both do the same thing but the former method invocation isn't available in the newly generated Scala classes.

**Comment by vitillo** [comment]
Even if tests run successfully don't merge it just yet; I would like to manually test it first on one of our jobs.

File: src/main/scala/com/mozilla/telemetry/heka/HekaFrame.scala (hunk 2 above)
Comment (maurodoglio) [inline src/main/scala/com/mozilla/telemetry/heka/HekaFrame.scala]: I C

**Comment by maurodoglio** [comment]
Reviewed 6 of 8 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/167)*

**Pull request merged**